### PR TITLE
Create Export-DbaSysDbUserObjects to replace the internal function

### PR DESCRIPTION
**Pull request by jpomfret**
## Type of Change

 - [X] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [X] New feature (non-breaking change, adds functionality)
 - [X] Breaking change (effects multiple commands or functionality)
 - [X] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [X] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [X] Documentation
 - [ ] Build system
 

### Purpose
Open for any feedback on this - it seems to work pretty well as much as I've tested it.

Changing from using internal function `Get-DbaSysDbUserObjectScript` in `Export-DbaInstance` to new function `Export-DbaSysDbUserObjects`.
Added pester tests for exporting:
 - tables
 - views
 - stored procs
 - triggers
 - table function
 - scalar function
 - rule

### Approach
- make new function to use either on its own or with Export-DbaInstance

### Commands to test
Examples in the script

### CODE REVIEW

**Comment by wsmelton** [comment]
- If we are cleaning up code to prevent duplication this should incorporate use of `Export-DbaScript` being that it does the same thing a portion of this function is doing.

- Why not just call this `Export-DbaDbUserObject` and have the functionality be generic so you just pull the type of object and it calls to `Export-DbaScript` to generate the output? So it is just a wrapper function and can be used for system or user objects.

- If you are removing the call to `Get-DbaSysDbUserObjectsScript` we need to delete that file as well in this PR. Nothing else was/is using it.

**Comment by jpomfret** [comment]
Hey @wsmelton,
I removed the file - wasn't sure whether I should have included that or not, thanks!

As for the other two comments I'm a little confused by what you mean, I am using `Export-DbaScript` within the new function to actually generate the files, just adding some logic to get the system objects depending on type.

File: functions/Export-DbaSysDbUserObject.ps1 [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                $scriptPath = Get-ExportFilePath -Path $PSBoundParameters.Path -FilePath $PSBoundParameters.FilePath -Type sql -ServerName $SessionObject.Instance
+
+                $systemDbs = "master", "model", "msdb"
+
+                foreach ($systemDb in $systemDbs) {
+                    $smoDb = $server.databases[$systemDb]
+                    $userObjects = @()
+                    $userObjects += $smoDb.Tables | Where-Object IsSystemObject -ne $true | Select-Object Name, @{l = 'SchemaName'; e = { $_.Schema } } , @{l = 'Type'; e = { 'TABLE' } }, @{l = 'Database'; e = { $systemDb } }
+                    $userObjects += $smoDb.Triggers | Where-Object IsSystemObject -ne $true | Select-Object Name, @{l = 'SchemaName'; e = { $null } } , @{l = 'Type'; e = { 'SQL_TRIGGER' } }, @{l = 'Database'; e = { $systemDb } }
+                    $userObjects += Get-DbaModule -SqlInstance $server -Database $systemDb -ExcludeSystemObjects -Type "View", "TableValuedFunction", "DefaultConstraint", "StoredProcedure", "Rule", "InlineTableValuedFunction", "ScalarFunction" | Sort-Object Type | Select-Object Name, SchemaName, Type, Database
```
Comment (wsmelton) [inline functions/Export-DbaSysDbUserObject.ps1]:
```suggestion
                    $params = @{
                        SqlInstance = $server
                        Database = $systemDb
                        ExcludeSystemObjects = $true
                        Type = 'View', 'TableValuedFunction', 'DefaultConstraint', 'StoredProcedure', 'Rule', 'InlineTableValuedFunction', 'ScalarFunction'
                    }
                    $userObjects += Get-DbaModule @params | Sort-Object Type | Select-Object Name, SchemaName, Type, Database
```

**Comment by potatoqualitee** [comment]
thanks much 🙇

**Pull request merged**